server/src/main/java/org/cloudfoundry/identity/uaa/oauth/UaaTokenStore.java

```diff
@@ -83,24 +84,36 @@ public class UaaTokenStore implements AuthorizationCodeServices {
     private final RandomValueStringGenerator generator = new RandomValueStringGenerator(32);
     private final RowMapper rowMapper = new TokenCodeRowMapper();
 
-    private AtomicReference<Instant> lastClean = new AtomicReference<>(Instant.EPOCH);
+    private final AtomicReference<Instant> lastClean = new AtomicReference<>(Instant.EPOCH);
+
+    private final Clock clock;
 
     public UaaTokenStore(DataSource dataSource) {
         this(dataSource, DEFAULT_EXPIRATION_TIME);
     }
 
     public UaaTokenStore(DataSource dataSource, Duration expirationTime) {
+        this(dataSource, expirationTime, Clock.systemDefaultZone());
```

Comment:
That's a good point. I don't think that I understand very much the difference between the two.

As far as I can tell, `Instant.now()` will return the same value, regardless of the clock. But then it may get mixed up if we ask for the date of the instant? 

I'm gonna keep reading a bit on it.

Reply:
I'm super confused.

To convert an instant to a date, you are forced to pass the zone again, which makes complete sense. That I understand.
```
Instant.now().atZone(ZoneId.systemDefault()).getDayOfMonth();
```

But then why is there a `Clock#getZone()` method. I quickly took a look at the library code, the zone doesn't appear to be used for anything in `Instant`. I'm wondering if we can just pass `null` here, as we are not going to use the Zone.

Reply:
I meant passing `null` for the ZoneId, but we don't have access to that constructor.

Reply:
Alright my quick investigation found that `Instant` doesn't use the Clock's zone for anything that we care about, so any System Clock will result in the same behavior.

I would like to do `Clock.system(null)` but there's a null check on the zone parameter, so we can't do that.

With all that information, I lean towards just leaving the code the way it is. Let me know what you think.

Reply:
Java 11 Instant.now() source code:
>     public static Instant now() {
>         return Clock.systemUTC().instant();
>     }
>

Reply:
Which is unbelievably confusing, because the `.instant()` call doesn't use the Clock's zone.